### ISSUE TITLE
Docs error in new `mistune` package

### ISSUE THREAD

**Issue by glenn-jocher** [ultralytics/ultralytics#18458]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Bug

Docs build error started today in new `mistune 3.1.0`:

```
INFO    -  Converting notebook (execute=False): /home/runner/work/ultralytics/ultralytics/docs/en/datasets/explorer/explorer.ipynb
ERROR   -  Error reading page 'datasets/explorer/explorer.ipynb': 'MathBlockParser' object has no attribute 'parse_axt_heading'
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.12.8/x64/bin/mkdocs", line 8, in <module>
    sys.exit(cli())
             ^^^^^
  File "/opt/hostedtoolcache/Python/3.12.8/x64/lib/python3.12/site-packages/click/core.py", line 1161, in __call__
    return self.main(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.12.8/x64/lib/python3.12/site-packages/click/core.py", line 1082, in main
    rv = self.invoke(ctx)
         ^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.12.8/x64/lib/python3.12/site-packages/click/core.py", line 1697, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.12.8/x64/lib/python3.12/site-packages/click/core.py", line 14[43](https://github.com/ultralytics/ultralytics/actions/runs/12539746413/job/34966190841#step:9:44), in invoke
    return ctx.invoke(self.callback, **ctx.params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.12.8/x64/lib/python3.12/site-packages/click/core.py", line 788, in invoke
    return __callback(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.12.8/x64/lib/python3.12/site-packages/mkdocs/__main__.py", line 288, in build_command
    build.build(cfg, dirty=not clean)
  File "/opt/hostedtoolcache/Python/3.12.8/x64/lib/python3.12/site-packages/mkdocs/commands/build.py", line 310, in build
    _populate_page(file.page, config, files, dirty)
  File "/opt/hostedtoolcache/Python/3.12.8/x64/lib/python3.12/site-packages/mkdocs/commands/build.py", line 167, in _populate_page
    page.render(config, files)
  File "/opt/hostedtoolcache/Python/3.12.8/x64/lib/python3.12/site-packages/mkdocs_jupyter/plugin.py", line 123, in new_render
    body = convert.nb2html(
           ^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.12.8/x64/lib/python3.12/site-packages/mkdocs_jupyter/nbconvert2.py", line 149, in nb2html
    content, resources = exporter.from_filename(
                         ^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.12.8/x64/lib/python3.12/site-packages/nbconvert/exporters/templateexporter.py", line 386, in from_filename
    return super().from_filename(filename, resources, **kw)  # type:ignore[return-value]
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.12.8/x64/lib/python3.12/site-packages/nbconvert/exporters/exporter.py", line 201, in from_filename
    return self.from_file(f, resources=resources, **kw)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.12.8/x64/lib/python3.12/site-packages/nbconvert/exporters/templateexporter.py", line 392, in from_file
    return super().from_file(file_stream, resources, **kw)  # type:ignore[return-value]
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.12.8/x64/lib/python3.12/site-packages/nbconvert/exporters/exporter.py", line 220, in from_file
    return self.from_notebook_node(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.12.8/x64/lib/python3.12/site-packages/nbconvert/exporters/html.py", line 268, in from_notebook_node
    html, resources = super().from_notebook_node(nb, resources, **kw)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.12.8/x64/lib/python3.12/site-packages/nbconvert/exporters/templateexporter.py", line 424, in from_notebook_node
    output = self.template.render(nb=nb_copy, resources=resources)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.12.8/x64/lib/python3.12/site-packages/jinja2/environment.py", line 1295, in render
    self.environment.handle_exception()
  File "/opt/hostedtoolcache/Python/3.12.8/x64/lib/python3.12/site-packages/jinja2/environment.py", line 942, in handle_exception
    raise rewrite_traceback_stack(source=source)
  File "/opt/hostedtoolcache/Python/3.12.8/x64/lib/python3.12/site-packages/mkdocs_jupyter/templates/mkdocs_html/notebook.html.j2", line 92, in top-level template code
    {%- set vars = {"cell_id": 0 } -%}
    ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.12.8/x64/share/jupyter/nbconvert/templates/lab/index.html.j2", line 4, in top-level template code
    {% from 'jupyter_widgets.html.j2' import jupyter_widgets %}
    ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.12.8/x64/share/jupyter/nbconvert/templates/lab/base.html.j2", line 3, in top-level template code
    {% from 'cell_id_anchor.j2' import cell_id_anchor %}
    ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.12.8/x64/share/jupyter/nbconvert/templates/base/display_priority.j2", line 1, in top-level template code
    {%- extends 'base/null.j2' -%}
    ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.12.8/x64/share/jupyter/nbconvert/templates/base/null.j2", line 26, in top-level template code
    {%- block body -%}
    ^^^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.12.8/x64/share/jupyter/nbconvert/templates/base/null.j2", line 29, in block 'body'
    {%- block body_loop -%}
^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.12.8/x64/share/jupyter/nbconvert/templates/base/null.j2", line 31, in block 'body_loop'
    {%- block any_cell scoped -%}
^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.12.8/x64/share/jupyter/nbconvert/templates/base/null.j2", line 87, in block 'any_cell'
    {%- block markdowncell scoped-%} {%- endblock markdowncell -%}
^^^^^
  File "/opt/hostedtoolcache/Python/3.12.8/x64/share/jupyter/nbconvert/templates/lab/base.html.j2", line 109, in block 'markdowncell'
    {%- set html_value=cell.source  | markdown2html | strip_files_prefix -%}
    ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.12.8/x64/lib/python3.12/site-packages/nbconvert/exporters/html.py", line 243, in markdown2html
    return MarkdownWithMath(renderer=renderer).render(source)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.12.8/x64/lib/python3.12/site-packages/nbconvert/filters/markdown_mistune.py", line [47](https://github.com/ultralytics/ultralytics/actions/runs/12539746413/job/34966190841#step:9:48)1, in __init__
    block = MathBlockParser()
            ^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.12.8/x64/lib/python3.12/site-packages/mistune/block_parser.py", line 111, in __init__
    name: getattr(self, 'parse_' + name) for name in self.SPECIFICATION
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'MathBlockParser' object has no attribute 'parse_axt_heading'. Did you mean: 'parse_atx_heading'?
Traceback (most recent call last):
  File "/home/runner/work/ultralytics/ultralytics/docs/build_docs.py", line 309, in <module>
Building docs from /home/runner/work/ultralytics/ultralytics/docs
    main()
  File "/home/runner/work/ultralytics/ultralytics/docs/build_docs.py", line 293, in main
    subprocess.run(f"mkdocs build -f {DOCS.parent}/mkdocs.yml --strict", check=True, shell=True)
  File "/opt/hostedtoolcache/Python/3.12.8/x64/lib/python3.12/subprocess.py", line [57](https://github.com/ultralytics/ultralytics/actions/runs/12539746413/job/34966190841#step:9:58)1, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command 'mkdocs build -f /home/runner/work/ultralytics/ultralytics/mkdocs.yml --strict' returned non-zero exit status 1.
```

### Environment

NA

### Minimal Reproducible Example

```
python docs/build_docs.py
```

### Additional

Linked to https://github.com/lepture/mistune/issues/403

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @glenn-jocher, thank you for bringing this up and for your detailed report 🚀. We understand your concern regarding the docs build issue related to `mistune 3.1.0`. Here's how we recommend proceeding:

### Upgrade Dependencies

First, ensure you're using the latest `ultralytics` package along with all its requirements. This ensures compatibility with the latest updates:

```bash
pip install -U ultralytics
```

For the dependency `mistune`, you might consider pinning a compatible version if this is a breaking change. You can do so by editing your `requirements.txt` or environment configuration.

### Minimal Reproducible Example (MRE)

We appreciate the MRE you've already provided 🙏:

```python
python docs/build_docs.py
```

This is very helpful for our debugging process! We will validate this and look into any issues.

### Related Discourse

Thank you also for linking the related issue in the `mistune` repository ([lepture/mistune#403](https://github.com/lepture/mistune/issues/403))! This context helps us and the wider community better understand the cause.

### Additional Recommendations

If you'd like to contribute troubleshooting steps, here's what we suggest:

1. Test the docs build process in an isolated Python environment, such as a virtualenv or conda, to rule out any conflicts with existing packages:
    ```bash
    python -m venv ultralytics-env
    source ultralytics-env/bin/activate     # On Windows, use `ultralytics-env\Scripts\activate`
    pip install -U ultralytics
    python docs/build_docs.py
    ```

2. Check your environment's Python version. We recommend using **Python>=3.8**.

3. Try downgrading `mistune` to an earlier compatible version (`<3.1.0`) as a potential quick fix:
    ```bash
    pip install mistune<3.1.0
    ```

---

### Community and Support

Join the Ultralytics community to discuss this or similar issues:

- Real-time chat on [Discord](https://discord.com/invite/ultralytics) 🎧
- In-depth discussions on [Discourse](https://community.ultralytics.com)
- Share insights with fellow users on our [Subreddit](https://reddit.com/r/Ultralytics)

---

### Status

We currently use automated continuous integration (CI) workflows to ensure `ultralytics` codebase integrity. You can always confirm the latest CI status here:  
<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

An Ultralytics engineer will take a closer look at this and assist you soon! Stay tuned 😃

**Comment by UltralyticsAssistant** [comment]
The potential fix for this issue has now been merged via [PR #18459](https://github.com/ultralytics/ultralytics/pull/18459), authored by @RizwanMunawar with contributions from @glenn-jocher. 🎉

### Overview of Key Changes  
This PR introduces significant updates to the documentation ecosystem by transitioning from Jupyter-based files to a markdown-based approach, enhancing accessibility and maintainability while streamlining workflows:  
- **Deprecated Jupyter-Based Documentation**: Removed Jupyter notebooks (`explorer.ipynb`) and related dependencies.  
- **Markdown-based Solution**: Introduced an improved `explorer.md` file for dataset exploration.  
- **Updated MkDocs Configuration**: Adjusted navigation and removed Jupyter-specific plugins to simplify and future-proof the documentation structure.  
- **Encouragement to Use Ultralytics HUB**: The ecosystem now leans toward the powerful, no-code features provided by Ultralytics HUB for data exploration, reducing reliance on notebooks.  

These changes aim to simplify maintenance, improve compatibility, and shift users to the more robust HUB features. While this transition improves long-term usability, users who previously relied on Jupyter-based docs may need some time to adjust.

### How to Test  
You can test the latest updates by running:  
```bash
pip install git+https://github.com/ultralytics/ultralytics.git@main
```
Alternatively, you can wait for the next official release on PyPI. If possible, please confirm whether these changes resolve the issue you reported.

Thank you so much for bringing this issue to our attention. Your input helps us make Ultralytics better for everyone! If the problem persists or you have additional feedback, feel free to share and we’ll be happy to assist. 🙏
